aptos-move/block-executor/src/scheduler.rs

```diff
@@ -165,14 +163,14 @@ impl ValidationStatus {
 
 pub struct Scheduler {
     /// Number of txns to execute, immutable.
-    num_txns: usize,
+    num_txns: TxnIndex,
 
     /// A shared index that tracks the minimum of all transaction indices that require execution.
     /// The threads increment the index and attempt to create an execution task for the corresponding
     /// transaction, if the status of the txn is 'ReadyToExecute'. This implements a counting-based
     /// concurrent ordered set. It is reduced as necessary when transactions become ready to be
     /// executed, in particular, when execution finishes and dependencies are resolved.
-    execution_idx: AtomicUsize,
+    execution_idx: AtomicU32,
```

Comment:
the atomic discussion reminds me, do we need to pad the two idx too?

Reply:
great point - I think why it wasn't padded is since all threads were reading the validation and execution indices pretty often, and then having them actually padded in separate cache-lines could be even worse (2x invalidations). Never did precise benchmarks though - so it's all about how much false sharing penalty vs actual sharing overhead of having them separate :-) But (a) we can definitely pad commit_idx and if and when we have an optimization for validator and executor distinction to stick to threads, it will actually become important.

So maybe let's pad all those indices and if there is no regression keep that way, more forward looking. I'll add a comment to validation & execution indices though regarding the above.